src/http/client/response.cr

```diff
@@ -131,14 +140,15 @@ class HTTP::Client::Response
       raise "Invalid HTTP status code: #{pieces[1]}"
     end
 
+    status = HTTP::Status.from_code(status_code)
     status_message = pieces[2]? ? pieces[2].chomp : ""
 
     body_type = HTTP::BodyType::OnDemand
-    body_type = HTTP::BodyType::Mandatory if mandatory_body?(status_code)
+    body_type = HTTP::BodyType::Mandatory if mandatory_body?(status)
```

Comment:
Maybe we could move this method to a `body_mandatory?` in the `HTTP::Status` enum (it could be done in a separate PR).

Reply:
Happy to tackle that one separately.